core/primitives/src/crypto.rs

```diff
@@ -419,11 +419,11 @@ impl<T: AsMut<[u8]> + AsRef<[u8]> + Default + Derive> Ss58Codec for T {
 		}
 		let ver = d[0].try_into().map_err(|_: ()| PublicError::UnknownVersion)?;
 
-		if d[len+1..len+3] != ss58hash(&d[0..len+1]).as_bytes()[0..2] {
+		if d[len+1..len+3] != ss58hash(&d[0..=len]).as_bytes()[0..2] {
```

Comment:
```suggestion
		if d[len+1..len+3] != ss58hash(&d[0..len+1]).as_bytes()[0..2] {
```

The `+1` is better because it matches the `+1` on the left side.